src/perfEvents_linux.cpp

```diff
@@ -187,7 +187,13 @@ void PerfEvents::installSignalHandler() {
 }
 
 void PerfEvents::signalHandler(int signo, siginfo_t* siginfo, void* ucontext) {
-    Profiler::_instance.recordSample(ucontext, 1, NULL);
+    u64 counter;
+    if (read(siginfo->si_fd, &counter, sizeof(counter)) != sizeof(counter)) {
```

Comment:
Could you please elaborate why is it needed here?

Reply:
read(siginfo->si_fd) reads the value of perf counter. I assume this read() should always succeed. `if` is left here just in case. Do you think it is better to ignore return value or handle it some other way?

Reply:
I think it's less confusing to handle it as `if (read(siginfo->si_fd, &counter, sizeof(counter)) == -1`

Reply:
Not sure this is better. I expect to read the counter variable fully - that's what I've written.